bookkeeper-server/src/main/java/org/apache/bookkeeper/auth/BookieAuthProvider.java

```diff
@@ -79,16 +75,19 @@ default void close() {}
     }
 
     /**
+     * Callback to let the provider know that the underlying protocol is changed.
```

Comment:
the AuthProvider on this callback has access to the Certificates.
The AuthProvider is instantiated when the connection is created, but no StartTLS as occurred yet, so at that time it is not possible to access client-side Certificates.
So this callback is useful to intercept the client SSL data and:
- schedule a close of the connection at the time of certificate expiration date
- prevent "forbidden" certificates to access Bookie data